cpu/stm32/cpu_init.c

```diff
@@ -160,6 +160,7 @@ static void _gpio_init_ain(void)
  * This very teniously avoids optimization, even optimized it's better than
  * nothing but periodic review should establish that it doesn't get optimized.
  */
+#if defined(STM32_OPTION_BYTES)
```

Comment:
either use attribute unused or move the #ifdef up from line 244 (not do the same #ifdef twice)

Reply:
I added a fixup that moves the #ifdef. I think attribute(unused) is wrong here. This is not three steps into linking or hidden behind 100 layers of dependency modelling. That function is exactly used three times and only when that option is enabled. An attribute does not convey that meaning/context but the ifdef does.